src/libtiled/tilelayer.cpp

```diff
@@ -594,7 +694,9 @@ TileLayer *TileLayer::clone() const
 TileLayer *TileLayer::initializeClone(TileLayer *clone) const
 {
     Layer::initializeClone(clone);
-    clone->mGrid = mGrid;
+    for (int x = 0; x < mWidth; ++x)
+        for (int y = 0; y < mHeight; ++y)
```

Comment:
In other cases you first iterate over the `y`, here you first iterate over `x`. Generally first iterating over `y` should be faster due to the access being `x + y * 16` regarding cache hits.